### PR TITLE
BUG: Unable to open Stata 118 or 119 format files saved in big-endian…

### LINKED ISSUE #58638: BUG: Unable to open Stata 118 or 119 files saved in big-endian format that contain strL data
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
# Both of the following lines fail (the data files are provided in the issue description)
df = pd.read_stata("stata12_be_118.dta")
df = pd.read_stata("stata12_be_119.dta")
```

### Issue Description

If I attempt to open a 118 format file saved in big-endian format that contains strL data I get the following error:
```python
>>> import pandas as pd
>>> df = pd.read_stata("stata12_be_118.dta")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Program Files\Python312\Lib\site-packages\pandas\io\stata.py", line 2109, in read_stata
    return reader.read()
           ^^^^^^^^^^^^^
  File "C:\Program Files\Python312\Lib\site-packages\pandas\io\stata.py", line 1775, in read
    data = self._insert_strls(data)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python312\Lib\site-packages\pandas\io\stata.py", line 1876, in _insert_strls
    data.isetitem(i, [self.GSO[str(k)] for k in data.iloc[:, i]])
                      ~~~~~~~~^^^^^^^^
KeyError: '844424930131969'
```

The same is true if I repeat this for a 119 format file:
```python
>>> import pandas as pd
>>> df = pd.read_stata("stata12_be_119.dta")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Program Files\Python312\Lib\site-packages\pandas\io\stata.py", line 2109, in read_stata
    return reader.read()
           ^^^^^^^^^^^^^
  File "C:\Program Files\Python312\Lib\site-packages\pandas\io\stata.py", line 1775, in read
    data = self._insert_strls(data)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\Python312\Lib\site-packages\pandas\io\stata.py", line 1876, in _insert_strls
    data.isetitem(i, [self.GSO[str(k)] for k in data.iloc[:, i]])
                      ~~~~~~~~^^^^^^^^
KeyError: '3298534883329'
```
The equivalent 117 format file works fine:
```python
>>> import pandas as pd
>>> df = pd.read_stata("stata12_be_117.dta")
>>> df
      x    y                  z
0   1.0  abc          abcdefghi
1   3.0  cba  qwertywertyqwerty
2  93.0                    strl
```

This occurs due to a failed lookup for the strL value due to a mismatch in the expected key for the following reason:

strL content is stored separately to the main data in records identified by a (v, o) - (variable, observation) value. This is then referenced from the main data to associate a particular string value with a position in the data. In format 117 v and o were both stored in 4 bytes and there was an exact match between the value stored in the main data and in the strL records. Format 118 and later increased o to be stored in 8 bytes, allowing more observations to be held in the data, however it did not change the storage size in the main data for referencing this, resulting in a need for a packed storage value where some of the high bytes were removed from v and o to allow both values to fit in 8 bytes. Using the notation of letters to represent bytes in v and numbers to represent bytes in o this means that the (v, o) index:

(ABCD, 12345678)
would be referenced in 118 by:
AB123456
and in 119 by:
ABC12345

In big-endian format:
(DCBA, 87654321)
would be reference in 118 by:
BA654321
and in 119 by:
CBA54321

When looking up values Pandas takes the approach of converting (v, o) in the strL records into the packed form and treating this as an 8-byte integer, rather than expanding out the values in the data into separate 4 and 8-byte integers. The current code branch for little-endian gives the expected result:

```python
>>> buf = 'ABCD12345678'
>>> v_size = 2 # 118 format
>>> buf[0:v_size] + buf[4 : (12 - v_size)]
'AB123456'
>>> v_size = 3 # 119 format
>>> buf[0:v_size] + buf[4 : (12 - v_size)]
'ABC12345'
```
however the big-endian path is incorrect:
```python
>>> buf = 'DCBA87654321'
>>> v_size = 2 # 118 format
>>> buf[0:v_size] + buf[(4 + v_size) :]
'DC654321'
>>> v_size = 3 # 119 format
>>> buf[0:v_size] + buf[(4 + v_size) :]
'DCB54321'
```

it should instead be:
```python
>>> buf = 'DCBA87654321'
>>> v_size = 2 # 118 format
>>> buf[4 - v_size:4] + buf[(4 + v_size) :]
'BA654321'
>>> v_size = 3 # 119 format
>>> buf[4 - v_size:4] + buf[(4 + v_size) :]
'CBA54321'
```

Once the packed value has been determined it should have byteorder applied, as this happens to the main data.

[stata12_be.zip](https://github.com/pandas-dev/pandas/files/15250866/stata12_be.zip)

### Expected Behavior

I would expect the file to load successfully, as it does in Stata:
```
. use "stata12_be_118.dta"

. list

     +------------------------------+
     |  x     y                   z |
     |------------------------------|
  1. |  1   abc           abcdefghi |
  2. |  3   cba   qwertywertyqwerty |
  3. | 93                      strl |
     +------------------------------+
```

```
. use "stata12_be_119.dta"

. list

     +------------------------------+
     |  x     y                   z |
     |------------------------------|
  1. |  1   abc           abcdefghi |
  2. |  3   cba   qwertywertyqwerty |
  3. | 93                      strl |
     +------------------------------+
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : d9cdd2ee5a58015ef6f4d15c7226110c9aab8140
python                : 3.12.3.final.0
python-bits           : 64
OS                    : Windows
OS-release            : 10
Version               : 10.0.19045
machine               : AMD64
processor             : Intel64 Family 6 Model 158 Stepping 10, GenuineIntel
byteorder             : little
LC_ALL                : None
LANG                  : None
LOCALE                : English_United Kingdom.1252

pandas                : 2.2.2
numpy                 : 1.26.4
pytz                  : 2024.1
dateutil              : 2.9.0.post0
setuptools            : 69.5.1
pip                   : 24.0
Cython                : None
pytest                : None
hypothesis            : None
sphinx                : None
blosc                 : None
feather               : None
xlsxwriter            : None
lxml.etree            : 5.2.1
html5lib              : 1.1
pymysql               : None
psycopg2              : None
jinja2                : None
IPython               : None
pandas_datareader     : None
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : 4.12.3
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : None
fsspec                : None
gcsfs                 : None
matplotlib            : 3.8.4
numba                 : None
numexpr               : 2.10.0
odfpy                 : None
openpyxl              : None
pandas_gbq            : None
pyarrow               : None
pyreadstat            : None
python-calamine       : None
pyxlsb                : None
s3fs                  : None
scipy                 : 1.13.0
sqlalchemy            : None
tables                : None
tabulate              : None
xarray                : None
xlrd                  : None
zstandard             : None
tzdata                : 2024.1
qtpy                  : None
pyqt5                 : None

</details>

### PR DESCRIPTION
**Pull request by cmjcharlton**
… format that contain strL data

- [x] closes #58638
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by bashtage** [comment]
Just to check - are the stata files produced by Stata?

File: pandas/tests/io/test_stata.py [hunk 1]
```diff
@@ -1800,8 +1821,18 @@ def test_gzip_writing(self, temp_file):
...
+    @pytest.mark.parametrize(
+        "file",
+        [
+            "stata16_118",
+            "stata16_be_118",
+            "stata16_119",
+            "stata16_be_119",
+        ],
+    )
+    def test_unicode_dta_118(self, file, datapath):
```
Comment (bashtage) [inline pandas/tests/io/test_stata.py:1834]: Maybe rename 118 -> 118_119

File: pandas/tests/io/test_stata.py [hunk 2]
```diff
@@ -331,8 +342,18 @@ def test_read_dta12(self, datapath):
...
+    @pytest.mark.parametrize(
+        "file",
+        [
+            "stata14_118",
+            "stata14_be_118",
+            "stata14_119",
+            "stata14_be_119",
+        ],
+    )
+    def test_read_dta18(self, file, datapath):
```
Comment (bashtage) [inline pandas/tests/io/test_stata.py:355]: Similar rename here would be clearer.

**Review by bashtage — CHANGES_REQUESTED** [review]
Small changes to clarify changed test. Otherwise LGTM.  Only other question is to verify dta files are created by Stata.

**Comment by cmjcharlton** [comment]
These files were not created with Stata but were made using the published specification. I have however checked that they all load correctly into Stata and have used it to verify that they are constructed correctly:

```
. use stata12_118.dta

. list

     +------------------------------+
     |  x     y                   z |
     |------------------------------|
  1. |  1   abc           abcdefghi |
  2. |  3   cba   qwertywertyqwerty |
  3. | 93                      strl |
     +------------------------------+

. dtaverify stata12_118.dta
  (file "stata12_118.dta" is .dta-format 118 from Stata 14, 15, 16, 17, or 18)

 1. reading and verifying header
    release is 118
    byteorder LSF
    K (# of vars) is 3
    N (# of obs) is 3                    
    label length 0 ||
    date length 17 |19 Feb 2015 09:20|

 2. reading and verifying map
    map[ 1] =                    0
    map[ 2] =                  158
    map[ 3] =                  281
    map[ 4] =                  320
    map[ 5] =                  728
    map[ 6] =                  757
    map[ 7] =                  947
    map[ 8] =                 1373
    map[ 9] =                 2371
    map[10] =                 2406
    map[11] =                 2473
    map[12] =                 2581
    map[13] =                 2610
    map[14] =                 2622
    verifying map
    1 2 3 4 5 6 7 8 9 10 11 12 13 14

 3. reading and verifying vartypes

 4. reading and verifying varnames
    verifying varnames unique

 5. reading and verifying sort order
    verifying contents

 6. reading and verifying display formats
    verifying formats
    verifying formats correspond to variable type

 7. reading and verifying value-label assignment
    verifying value-label construction
    verifying value label and corresponding variable types

 8. reading and verifying variable-label assignment
    verifying variable labels construction

 9. reading and verifying characteristics
    (0 characteristics in file)

10. reading and verifying data
    (dtaverify_118 cannot verify that values are correct)
    verifying construction
    (1 strL variables)
    verifying strL construction

11. reading and verifying strLs
    verifying construction
    (3 strLs expected)

12. reading and verifying value label definitions
    verifying construction
    (0 labels in file)

no errors or warnings above; .dta file valid

. use stata12_119.dta

. list

     +------------------------------+
     |  x     y                   z |
     |------------------------------|
  1. |  1   abc           abcdefghi |
  2. |  3   cba   qwertywertyqwerty |
  3. | 93                      strl |
     +------------------------------+

. dtaverify stata12_119.dta
  (file "stata12_119.dta" is .dta-format 119 from Stata 15, 16, 17, or 18)

 1. reading and verifying header
    release is 119
    byteorder LSF
    K (# of vars) is 3
    N (# of obs) is 3                    
    label length 0 ||
    date length 17 |19 Feb 2015 09:20|

 2. reading and verifying map
    map[ 1] =                    0
    map[ 2] =                  160
    map[ 3] =                  283
    map[ 4] =                  322
    map[ 5] =                  730
    map[ 6] =                  767
    map[ 7] =                  957
    map[ 8] =                 1383
    map[ 9] =                 2381
    map[10] =                 2416
    map[11] =                 2483
    map[12] =                 2591
    map[13] =                 2620
    map[14] =                 2632
    verifying map
    1 2 3 4 5 6 7 8 9 10 11 12 13 14

 3. reading and verifying vartypes

 4. reading and verifying varnames
    verifying varnames unique

 5. reading and verifying sort order
    verifying contents

 6. reading and verifying display formats
    verifying formats
    verifying formats correspond to variable type

 7. reading and verifying value-label assignment
    verifying value-label construction
    verifying value label and corresponding variable types

 8. reading and verifying variable-label assignment
    verifying variable labels construction

 9. reading and verifying characteristics
    (0 characteristics in file)

10. reading and verifying data
    (dtaverify_119 cannot verify that values are correct)
    verifying construction
    (1 strL variables)
    verifying strL construction

11. reading and verifying strLs
    verifying construction
    (3 strLs expected)

12. reading and verifying value label definitions
    verifying construction
    (0 labels in file)

no errors or warnings above; .dta file valid

. use stata12_be_117.dta

. list

     +------------------------------+
     |  x     y                   z |
     |------------------------------|
  1. |  1   abc           abcdefghi |
  2. |  3   cba   qwertywertyqwerty |
  3. | 93                      strl |
     +------------------------------+

. dtaverify stata12_be_117.dta
  (file "stata12_be_117.dta" is .dta-format 117 from Stata 13)

 1. reading and verifying header
    release is 117
    byteorder MSF
    K (# of vars) is 3
    N (# of obs) is 3
    label length 0 ||
    date length 17 |19 Feb 2015 09:20|

 2. reading and verifying map
    map[ 1] =                    0
    map[ 2] =                  153
    map[ 3] =                  276
    map[ 4] =                  315
    map[ 5] =                  435
    map[ 6] =                  464
    map[ 7] =                  630
    map[ 8] =                  768
    map[ 9] =                 1046
    map[10] =                 1081
    map[11] =                 1148
    map[12] =                 1244
    map[13] =                 1273
    map[14] =                 1285
    verifying map
    1 2 3 4 5 6 7 8 9 10 11 12 13 14

 3. reading and verifying vartypes

 4. reading and verifying varnames
    verifying varnames unique

 5. reading and verifying sort order
    verifying contents

 6. reading and verifying display formats
    verifying formats
    verifying formats correspond to variable type

 7. reading and verifying value-label assignment
    verifying value-label construction
    verifying value label and corresponding variable types

 8. reading and verifying variable-label assignment
    verifying variable labels construction

 9. reading and verifying characteristics
    (0 characteristics in file)

10. reading and verifying data
    (dtaverify_117 cannot verify that values are correct)
    verifying construction
    (1 strL variables)
    verifying strL construction

11. reading and verifying strLs
    verifying construction
    (3 strLs expected)

12. reading and verifying value label definitions
    verifying construction
    (0 labels in file)

no errors or warnings above; .dta file valid

. use stata12_be_118.dta

. list

     +------------------------------+
     |  x     y                   z |
     |------------------------------|
  1. |  1   abc           abcdefghi |
  2. |  3   cba   qwertywertyqwerty |
  3. | 93                      strl |
     +------------------------------+

. dtaverify stata12_be_118.dta
  (file "stata12_be_118.dta" is .dta-format 118 from Stata 14, 15, 16, 17, or 18)

 1. reading and verifying header
    release is 118
    byteorder MSF
    K (# of vars) is 3
    N (# of obs) is 3                    
    label length 0 ||
    date length 17 |19 Feb 2015 09:20|

 2. reading and verifying map
    map[ 1] =                    0
    map[ 2] =                  158
    map[ 3] =                  281
    map[ 4] =                  320
    map[ 5] =                  728
    map[ 6] =                  757
    map[ 7] =                  947
    map[ 8] =                 1373
    map[ 9] =                 2371
    map[10] =                 2406
    map[11] =                 2473
    map[12] =                 2581
    map[13] =                 2610
    map[14] =                 2622
    verifying map
    1 2 3 4 5 6 7 8 9 10 11 12 13 14

 3. reading and verifying vartypes

 4. reading and verifying varnames
    verifying varnames unique

 5. reading and verifying sort order
    verifying contents

 6. reading and verifying display formats
    verifying formats
    verifying formats correspond to variable type

 7. reading and verifying value-label assignment
    verifying value-label construction
    verifying value label and corresponding variable types

 8. reading and verifying variable-label assignment
    verifying variable labels construction

 9. reading and verifying characteristics
    (0 characteristics in file)

10. reading and verifying data
    (dtaverify_118 cannot verify that values are correct)
    verifying construction
    (1 strL variables)
    verifying strL construction

11. reading and verifying strLs
    verifying construction
    (3 strLs expected)

12. reading and verifying value label definitions
    verifying construction
    (0 labels in file)

no errors or warnings above; .dta file valid

. use stata12_be_119.dta

. list

     +------------------------------+
     |  x     y                   z |
     |------------------------------|
  1. |  1   abc           abcdefghi |
  2. |  3   cba   qwertywertyqwerty |
  3. | 93                      strl |
     +------------------------------+

. dtaverify stata12_be_119.dta
  (file "stata12_be_119.dta" is .dta-format 119 from Stata 15, 16, 17, or 18)

 1. reading and verifying header
    release is 119
    byteorder MSF
    K (# of vars) is 3
    N (# of obs) is 3                    
    label length 0 ||
    date length 17 |19 Feb 2015 09:20|

 2. reading and verifying map
    map[ 1] =                    0
    map[ 2] =                  160
    map[ 3] =                  283
    map[ 4] =                  322
    map[ 5] =                  730
    map[ 6] =                  767
    map[ 7] =                  957
    map[ 8] =                 1383
    map[ 9] =                 2381
    map[10] =                 2416
    map[11] =                 2483
    map[12] =                 2591
    map[13] =                 2620
    map[14] =                 2632
    verifying map
    1 2 3 4 5 6 7 8 9 10 11 12 13 14

 3. reading and verifying vartypes

 4. reading and verifying varnames
    verifying varnames unique

 5. reading and verifying sort order
    verifying contents

 6. reading and verifying display formats
    verifying formats
    verifying formats correspond to variable type

 7. reading and verifying value-label assignment
    verifying value-label construction
    verifying value label and corresponding variable types

 8. reading and verifying variable-label assignment
    verifying variable labels construction

 9. reading and verifying characteristics
    (0 characteristics in file)

10. reading and verifying data
    (dtaverify_119 cannot verify that values are correct)
    verifying construction
    (1 strL variables)
    verifying strL construction

11. reading and verifying strLs
    verifying construction
    (3 strLs expected)

12. reading and verifying value label definitions
    verifying construction
    (0 labels in file)

no errors or warnings above; .dta file valid

. use stata14_119.dta
(This is a  Ünicode data label)

. list

     +---------------------------------------------------------------------------+
     | Things   Cities   Unicode~l   Ints   Floats              Bytes      Longs |
     |---------------------------------------------------------------------------|
  1. |    Cat   Bogota      Bogotá      1        1   option b Ünicode          1 |
  2. |    Dog   Boston   Uzunköprü      .        .                  .          . |
  3. |  Plane     Rome      Tromsø      0        0           option a          0 |
  4. | Potato    Tokyo      Elâzığ     -4        4                  4          4 |
  5. |                                  0    .3333           option a   .3333333 |
     +---------------------------------------------------------------------------+

. dtaverify stata14_119.dta
  (file "stata14_119.dta" is .dta-format 119 from Stata 15, 16, 17, or 18)

 1. reading and verifying header
    release is 119
    byteorder LSF
    K (# of vars) is 7
    N (# of obs) is 5                    
    label length 30 |This is a  Ünicode data label|
    date length 17 |28 Apr 2015 17:58|

 2. reading and verifying map
    map[ 1] =                    0
    map[ 2] =                  190
    map[ 3] =                  313
    map[ 4] =                  360
    map[ 5] =                 1284
    map[ 6] =                 1337
    map[ 7] =                 1755
    map[ 8] =                 2697
    map[ 9] =                 4979
    map[10] =                 5014
    map[11] =                 5202
    map[12] =                 5335
    map[13] =                 5562
    map[14] =                 5574
    verifying map
    1 2 3 4 5 6 7 8 9 10 11 12 13 14

 3. reading and verifying vartypes

 4. reading and verifying varnames
    verifying varnames unique

 5. reading and verifying sort order
    verifying contents

 6. reading and verifying display formats
    verifying formats
    verifying formats correspond to variable type

 7. reading and verifying value-label assignment
    verifying value-label construction
    verifying value label and corresponding variable types

 8. reading and verifying variable-label assignment
    verifying variable labels construction

 9. reading and verifying characteristics
    (0 characteristics in file)

10. reading and verifying data
    (dtaverify_119 cannot verify that values are correct)
    verifying construction
    (1 strL variables)
    verifying strL construction

11. reading and verifying strLs
    verifying construction
    (4 strLs expected)

12. reading and verifying value label definitions
    verifying construction
    (1 labels in file)

no errors or warnings above; .dta file valid

. use stata14_be_118.dta
(This is a  Ünicode data label)

. list

     +---------------------------------------------------------------------------+
     | Things   Cities   Unicode~l   Ints   Floats              Bytes      Longs |
     |---------------------------------------------------------------------------|
  1. |    Cat   Bogota      Bogotá      1        1   option b Ünicode          1 |
  2. |    Dog   Boston   Uzunköprü      .        .                  .          . |
  3. |  Plane     Rome      Tromsø      0        0           option a          0 |
  4. | Potato    Tokyo      Elâzığ     -4        4                  4          4 |
  5. |                                  0    .3333           option a   .3333333 |
     +---------------------------------------------------------------------------+

. dtaverify stata14_be_118.dta
  (file "stata14_be_118.dta" is .dta-format 118 from Stata 14, 15, 16, 17, or 18)

 1. reading and verifying header
    release is 118
    byteorder MSF
    K (# of vars) is 7
    N (# of obs) is 5                    
    label length 30 |This is a  Ünicode data label|
    date length 17 |28 Apr 2015 17:58|

 2. reading and verifying map
    map[ 1] =                    0
    map[ 2] =                  188
    map[ 3] =                  311
    map[ 4] =                  358
    map[ 5] =                 1282
    map[ 6] =                 1319
    map[ 7] =                 1737
    map[ 8] =                 2679
    map[ 9] =                 4961
    map[10] =                 4996
    map[11] =                 5184
    map[12] =                 5317
    map[13] =                 5544
    map[14] =                 5556
    verifying map
    1 2 3 4 5 6 7 8 9 10 11 12 13 14

 3. reading and verifying vartypes

 4. reading and verifying varnames
    verifying varnames unique

 5. reading and verifying sort order
    verifying contents

 6. reading and verifying display formats
    verifying formats
    verifying formats correspond to variable type

 7. reading and verifying value-label assignment
    verifying value-label construction
    verifying value label and corresponding variable types

 8. reading and verifying variable-label assignment
    verifying variable labels construction

 9. reading and verifying characteristics
    (0 characteristics in file)

10. reading and verifying data
    (dtaverify_118 cannot verify that values are correct)
    verifying construction
    (1 strL variables)
    verifying strL construction

11. reading and verifying strLs
    verifying construction
    (4 strLs expected)

12. reading and verifying value label definitions
    verifying construction
    (1 labels in file)

no errors or warnings above; .dta file valid

. use stata14_be_119.dta
(This is a  Ünicode data label)

. list

     +---------------------------------------------------------------------------+
     | Things   Cities   Unicode~l   Ints   Floats              Bytes      Longs |
     |---------------------------------------------------------------------------|
  1. |    Cat   Bogota      Bogotá      1        1   option b Ünicode          1 |
  2. |    Dog   Boston   Uzunköprü      .        .                  .          . |
  3. |  Plane     Rome      Tromsø      0        0           option a          0 |
  4. | Potato    Tokyo      Elâzığ     -4        4                  4          4 |
  5. |                                  0    .3333           option a   .3333333 |
     +---------------------------------------------------------------------------+

. dtaverify stata14_be_119.dta
  (file "stata14_be_119.dta" is .dta-format 119 from Stata 15, 16, 17, or 18)

 1. reading and verifying header
    release is 119
    byteorder MSF
    K (# of vars) is 7
    N (# of obs) is 5                    
    label length 30 |This is a  Ünicode data label|
    date length 17 |28 Apr 2015 17:58|

 2. reading and verifying map
    map[ 1] =                    0
    map[ 2] =                  190
    map[ 3] =                  313
    map[ 4] =                  360
    map[ 5] =                 1284
    map[ 6] =                 1337
    map[ 7] =                 1755
    map[ 8] =                 2697
    map[ 9] =                 4979
    map[10] =                 5014
    map[11] =                 5202
    map[12] =                 5335
    map[13] =                 5562
    map[14] =                 5574
    verifying map
    1 2 3 4 5 6 7 8 9 10 11 12 13 14

 3. reading and verifying vartypes

 4. reading and verifying varnames
    verifying varnames unique

 5. reading and verifying sort order
    verifying contents

 6. reading and verifying display formats
    verifying formats
    verifying formats correspond to variable type

 7. reading and verifying value-label assignment
    verifying value-label construction
    verifying value label and corresponding variable types

 8. reading and verifying variable-label assignment
    verifying variable labels construction

 9. reading and verifying characteristics
    (0 characteristics in file)

10. reading and verifying data
    (dtaverify_119 cannot verify that values are correct)
    verifying construction
    (1 strL variables)
    verifying strL construction

11. reading and verifying strLs
    verifying construction
    (4 strLs expected)

12. reading and verifying value label definitions
    verifying construction
    (1 labels in file)

no errors or warnings above; .dta file valid

. use stata16_119.dta

. list

     +-----------------------------------------------+
     |   utf8   latin1   ascii   utf8_s~l   ascii_~l |
     |-----------------------------------------------|
  1. | ραηδας   PÄNDÄS       p     ραηδας          p |
  2. | ƤĀńĐąŜ        Ö       a     ƤĀńĐąŜ          a |
  3. | ᴘᴀᴎᴅᴀS        Ü       n     ᴘᴀᴎᴅᴀS          n |
  4. |                       d                     d |
  5. |                       a                     a |
     |-----------------------------------------------|
  6. |                       s                     s |
  7. |                                               |
     +-----------------------------------------------+

. dtaverify stata16_119.dta
  (file "stata16_119.dta" is .dta-format 119 from Stata 15, 16, 17, or 18)

 1. reading and verifying header
    release is 119
    byteorder LSF
    K (# of vars) is 5
    N (# of obs) is 7                    
    label length 0 ||
    date length 17 |31 May 2018 23:59|

 2. reading and verifying map
    map[ 1] =                    0
    map[ 2] =                  160
    map[ 3] =                  283
    map[ 4] =                  326
    map[ 5] =                  992
    map[ 6] =                 1037
    map[ 7] =                 1341
    map[ 8] =                 2025
    map[ 9] =                 3665
    map[10] =                 3700
    map[11] =                 4266
    map[12] =                 4587
    map[13] =                 4616
    map[14] =                 4628
    verifying map
    1 2 3 4 5 6 7 8 9 10 11 12 13 14

 3. reading and verifying vartypes

 4. reading and verifying varnames
    verifying varnames unique

 5. reading and verifying sort order
    verifying contents

 6. reading and verifying display formats
    verifying formats
    verifying formats correspond to variable type

 7. reading and verifying value-label assignment
    verifying value-label construction
    verifying value label and corresponding variable types

 8. reading and verifying variable-label assignment
    verifying variable labels construction

 9. reading and verifying characteristics
    (0 characteristics in file)

10. reading and verifying data
    (dtaverify_119 cannot verify that values are correct)
    verifying construction
    (2 strL variables)
    verifying strL construction

11. reading and verifying strLs
    verifying construction
    (12 strLs expected)

12. reading and verifying value label definitions
    verifying construction
    (0 labels in file)

no errors or warnings above; .dta file valid

. use stata16_be_118.dta

. list

     +-----------------------------------------------+
     |   utf8   latin1   ascii   utf8_s~l   ascii_~l |
     |-----------------------------------------------|
  1. | ραηδας   PÄNDÄS       p     ραηδας          p |
  2. | ƤĀńĐąŜ        Ö       a     ƤĀńĐąŜ          a |
  3. | ᴘᴀᴎᴅᴀS        Ü       n     ᴘᴀᴎᴅᴀS          n |
  4. |                       d                     d |
  5. |                       a                     a |
     |-----------------------------------------------|
  6. |                       s                     s |
  7. |                                               |
     +-----------------------------------------------+

. dtaverify stata16_be_118.dta
  (file "stata16_be_118.dta" is .dta-format 118 from Stata 14, 15, 16, 17, or 18)

 1. reading and verifying header
    release is 118
    byteorder MSF
    K (# of vars) is 5
    N (# of obs) is 7                    
    label length 0 ||
    date length 17 |31 May 2018 23:59|

 2. reading and verifying map
    map[ 1] =                    0
    map[ 2] =                  158
    map[ 3] =                  281
    map[ 4] =                  324
    map[ 5] =                  990
    map[ 6] =                 1023
    map[ 7] =                 1327
    map[ 8] =                 2011
    map[ 9] =                 3651
    map[10] =                 3686
    map[11] =                 4252
    map[12] =                 4573
    map[13] =                 4602
    map[14] =                 4614
    verifying map
    1 2 3 4 5 6 7 8 9 10 11 12 13 14

 3. reading and verifying vartypes

 4. reading and verifying varnames
    verifying varnames unique

 5. reading and verifying sort order
    verifying contents

 6. reading and verifying display formats
    verifying formats
    verifying formats correspond to variable type

 7. reading and verifying value-label assignment
    verifying value-label construction
    verifying value label and corresponding variable types

 8. reading and verifying variable-label assignment
    verifying variable labels construction

 9. reading and verifying characteristics
    (0 characteristics in file)

10. reading and verifying data
    (dtaverify_118 cannot verify that values are correct)
    verifying construction
    (2 strL variables)
    verifying strL construction

11. reading and verifying strLs
    verifying construction
    (12 strLs expected)

12. reading and verifying value label definitions
    verifying construction
    (0 labels in file)

no errors or warnings above; .dta file valid

. use stata16_be_119.dta

. list

     +-----------------------------------------------+
     |   utf8   latin1   ascii   utf8_s~l   ascii_~l |
     |-----------------------------------------------|
  1. | ραηδας   PÄNDÄS       p     ραηδας          p |
  2. | ƤĀńĐąŜ        Ö       a     ƤĀńĐąŜ          a |
  3. | ᴘᴀᴎᴅᴀS        Ü       n     ᴘᴀᴎᴅᴀS          n |
  4. |                       d                     d |
  5. |                       a                     a |
     |-----------------------------------------------|
  6. |                       s                     s |
  7. |                                               |
     +-----------------------------------------------+

. dtaverify stata16_be_119.dta
  (file "stata16_be_119.dta" is .dta-format 119 from Stata 15, 16, 17, or 18)

 1. reading and verifying header
    release is 119
    byteorder MSF
    K (# of vars) is 5
    N (# of obs) is 7                    
    label length 0 ||
    date length 17 |31 May 2018 23:59|

 2. reading and verifying map
    map[ 1] =                    0
    map[ 2] =                  160
    map[ 3] =                  283
    map[ 4] =                  326
    map[ 5] =                  992
    map[ 6] =                 1037
    map[ 7] =                 1341
    map[ 8] =                 2025
    map[ 9] =                 3665
    map[10] =                 3700
    map[11] =                 4266
    map[12] =                 4587
    map[13] =                 4616
    map[14] =                 4628
    verifying map
    1 2 3 4 5 6 7 8 9 10 11 12 13 14

 3. reading and verifying vartypes

 4. reading and verifying varnames
    verifying varnames unique

 5. reading and verifying sort order
    verifying contents

 6. reading and verifying display formats
    verifying formats
    verifying formats correspond to variable type

 7. reading and verifying value-label assignment
    verifying value-label construction
    verifying value label and corresponding variable types

 8. reading and verifying variable-label assignment
    verifying variable labels construction

 9. reading and verifying characteristics
    (0 characteristics in file)

10. reading and verifying data
    (dtaverify_119 cannot verify that values are correct)
    verifying construction
    (2 strL variables)
    verifying strL construction

11. reading and verifying strLs
    verifying construction
    (12 strLs expected)

12. reading and verifying value label definitions
    verifying construction
    (0 labels in file)

no errors or warnings above; .dta file valid
```

File: pandas/tests/io/test_stata.py (hunk 1 above)
Comment (cmjcharlton) [reply pandas/tests/io/test_stata.py]: Good idea - I have now renamed these functions.

File: pandas/tests/io/test_stata.py (hunk 2 above)
Comment (cmjcharlton) [reply pandas/tests/io/test_stata.py]: I have now renamed this as suggested.

**Comment by bashtage** [comment]
Can you ping on green and I'll merge this in.

**Review by bashtage — APPROVED** [review]

**Comment by cmjcharlton** [comment]
Thanks @bashtage . All checks how how completed.

**Comment by bashtage** [comment]
Thanks for the fix.

**Pull request merged**